File: apps/web/lib/retrieval.ts
export type RetrievalFilters = {
  source_types?: string[];
  source_ids?: string[];
  document_ids?: string[];
};

export type Citation = {
  citation_id: string;
  source_id: string;
  document_id: string;
  document_version: number;
  chunk_id: string;
  title: string;
  source_display_name: string;
  source_type: string;
  source_uri: string | null;
  section_path: string | null;
  page_number: number | null;
  excerpt: string;
  final_rank: number;
};

File: apps/api/app/schemas/retrieval.py
from uuid import UUID

from pydantic import BaseModel, Field


class RetrievalFiltersRequest(BaseModel):
    source_types: list[str] = Field(default_factory=list)
    source_ids: list[UUID] = Field(default_factory=list)
    document_ids: list[UUID] = Field(default_factory=list)


class CitationResponse(BaseModel):
    citation_id: str
    source_id: UUID
    document_id: UUID
    document_version: int
    chunk_id: UUID
    title: str
    source_display_name: str
    source_type: str
    source_uri: str | None
    section_path: str | None
    page_number: int | None
    excerpt: str
    final_rank: int
